test/e2e/tests/00-begin.js

```diff
@@ -4,44 +4,76 @@ const config = require('../../../config');
 // FIXME: This should come in from a config variable
 const url = 'http://localhost:' + config.port; // eslint-disable-line
 
+// Test timeout constants
+// TODO(awong): Move to a common file usable for all tests.
+const defaultTimeout = 500;   // The normal timeout to use. For most opreations w/o a server roundtrip, this should be more than fast enough.
+const slowTimeout = 1000;     // A slow timeout incase the page is doing something complex.
+const molassesTimeout = 5000; // A really really slow timeout. This should rarely be used.
+
+// TODO(awong): Move this into a custom command or assertion that can be used with client.expect.element().
+function expectNavigateAwayFrom(client, url) {
+    client.expect.element('.js-test-location').attribute('data-location')
+      .to.not.contain(url).before(defaultTimeout);
+}
+
 module.exports = {
   'Begin application': (client) => {
     console.log(url);
+
+    // Ensure introduction page renders.
     client
       .url(url)
-      .waitForElementVisible('body', 1000)
+      .waitForElementVisible('body', defaultTimeout)
       .assert.title('Apply for Health Care: Vets.gov')
-      .waitForElementVisible('.form-panel', 3000)
-      .click('.form-panel .usa-button-primary')
-      .pause(1000)
-      .assert.visible('input[name="fname"]')
+      .waitForElementVisible('.form-panel', slowTimeout)  // First render of React may be slow.
+      .click('.form-panel .usa-button-primary');
+    expectNavigateAwayFrom(client, '/introduction');
+
+    // Personal Information page.
+    client.expect.element('input[name="fname"]').to.be.visible;
+    client
       .setValue('input[name="fname"]', 'William')
       .setValue('input[name="lname"]', 'Shakespeare')
-      .click('.form-panel .usa-button-primary')
-      .pause(1000)
-      .assert.visible('select[name="veteranBirthMonth"]')
+      .click('.form-panel .usa-button-primary');
+    expectNavigateAwayFrom(client, '/veteran-information/personal-information');
```

Comment:
Makes sense (also love the concept of stanzas). I was just thinking that validating that our basic environment is sane (#1) by checking if one particular input is visible seems incomplete. And perhaps also validating that the environment is sane by checking the url would be more complete? But I see how it's adding 2 separate checks for #1, which muddies the process.